crates/autopilot/src/domain/competition/auction_mechanism/single_winner.rs

```diff
@@ -0,0 +1,225 @@
+use {
+    super::{AuctionMechanism, ComputedScores, NoWinners, Participant, ReferenceScore, Unranked},
+    crate::{
+        domain::{Auction, competition::TradedOrder},
+        infra,
+    },
+    primitive_types::U256,
+    rand::seq::SliceRandom,
+    std::collections::{HashMap, HashSet},
+};
+
+pub struct SingleWinnerAuctionMechanism {
+    pub eth: infra::Ethereum,
+    pub max_solutions_per_solver: usize,
+    pub max_winners_per_auction: usize,
```

Comment:
Having a `max_winners_per_auction` on a type which implies there can only be one winner is odd.
Our current logic technically already supports multiple winners. The difference to combinatorial auctions will therefor not be the number of winners but rather how solutions get picked and scores get computed.
So I guess the concrete suggestion here would be to change the name to something less contradictory.
I believe the core of the current logic is to pick multiple winners where no traded tokens ever overlap.